src/integrations/blocks/structured-data-blocks.php

```diff
@@ -56,4 +122,97 @@ public function enqueue_block_editor_assets() {
 		$this->asset_manager->enqueue_script( 'structured-data-blocks' );
 		$this->asset_manager->enqueue_style( 'structured-data-blocks' );
 	}
+
+	/**
+	 * Optimizes images in the FAQ blocks.
+	 *
+	 * @param array  $attributes The attributes.
+	 * @param string $content    The content.
+	 *
+	 * @return string The content with images optimized.
+	 */
+	public function optimize_faq_images( $attributes, $content ) {
+		if ( ! isset( $attributes['questions'] ) ) {
+			return $content;
+		}
+
+		return $this->optimize_images( $attributes['questions'], 'answer', $content );
+	}
+
+	/**
+	 * Optimizes images in the How-To blocks.
+	 *
+	 * @param array  $attributes The attributes.
+	 * @param string $content    The content.
+	 *
+	 * @return string The content with images optimized.
+	 */
+	public function optimize_how_to_images( $attributes, $content ) {
+		if ( ! isset( $attributes['steps'] ) ) {
+			return $content;
+		}
+
+		return $this->optimize_images( $attributes['steps'], 'text', $content );
+	}
+
+	/**
+	 * Optimizes images in structured data blocks.
+	 *
+	 * @param array  $data    The data from the attributes.
+	 * @param string $key     The key in the data to iterate over.
+	 * @param string $content The content.
+	 *
+	 * @return string The content with images optimized.
+	 */
+	private function optimize_images( $data, $key, $content ) {
+		// First grab all image IDs from the attributes.
+		$images = [];
+		foreach ( $data as $question ) {
```

Comment:
It's not a `$question` in the context of how to blocks. Though, I can't really think of a good alternative name without being super wordy, yet vague.
```suggestion
		foreach ( $block_attribute_data_list as $block_attribute_data ) {
```